packages/client-node/src/connection/node_https_connection.ts

```diff
@@ -24,10 +27,24 @@ export class NodeHttpsConnection extends NodeBaseConnection {
     params?: BaseQueryParams,
   ): Http.OutgoingHttpHeaders {
     if (this.params.tls !== undefined) {
-      const headers: Http.OutgoingHttpHeaders = {
-        ...this.defaultHeaders,
-        'X-ClickHouse-User': params?.auth?.username ?? this.params.username,
-        'X-ClickHouse-Key': params?.auth?.password ?? this.params.password,
+      if (this.params.auth.type === 'JWT') {
+        throw new Error(
+          'JWT auth is not supported with HTTPS connection using custom certificates',
+        )
+      }
+      let headers: Http.OutgoingHttpHeaders
+      if (isCredentialsAuth(params?.auth)) {
```

Comment:
It's not the same behavior. With the current change, both params must be provided. I'm okay with it. Let's just add it to the changelog